src/main/java/com/github/galatynf/sihywtcamd/mixin/LivingEntityMixin.java

```diff
@@ -30,27 +31,33 @@ public abstract class LivingEntityMixin extends Entity {
     public LivingEntityMixin(EntityType<?> type, World world) {
         super(type, world);
     }
-
+    @Unique public float xlastDamageTaken;
+    @Unique public float xtimeUntilRegen;
     @Shadow protected float lastDamageTaken;
     @Shadow public float bodyYaw;
 
     @Shadow public abstract boolean isBaby();
     @Shadow @Nullable public abstract LivingEntity getAttacker();
     @Shadow @Nullable public abstract EntityAttributeInstance getAttributeInstance(RegistryEntry<EntityAttribute> attribute);
 
+    @Shadow public abstract void setHealth(float health);
+
     @Inject(method = "computeFallDamage", at = @At("HEAD"), cancellable = true)
-    private void cancelArthropodFallDamage(float fallDistance, float damageMultiplier, CallbackInfoReturnable<Integer> cir) {
+    private void cancelArthropodFallDamage(double fallDistance, float damageMultiplier, CallbackInfoReturnable<Integer> cir) {
         if (ModConfig.get().arthropods.general.noFallDamage && this.getType().isIn(EntityTypeTags.ARTHROPOD)) {
             cir.setReturnValue(0);
         }
     }
-
-    @Inject(method = "damage", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/LimbAnimator;setSpeed(F)V"))
-    private void healUndeadAttacker(ServerWorld world, DamageSource source, float amount,
-                                    CallbackInfoReturnable<Boolean> cir) {
+    @Inject(method = "damage", at = @At("HEAD"))
+    private void copyValues(ServerWorld world, DamageSource source, float amount, CallbackInfoReturnable<Boolean> cir){
+        xlastDamageTaken = this.lastDamageTaken;
+        xtimeUntilRegen = this.timeUntilRegen;
+    }
+    @Inject(method = "damage", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/LivingEntity;becomeAngry(Lnet/minecraft/entity/damage/DamageSource;)V"))
+    private void healUndeadAttacker(ServerWorld world, DamageSource source, float amount, CallbackInfoReturnable<Boolean> cir) {
         if (ModConfig.get().undead.general.attackHeal && amount > 0) {
-            float realAmount = (float) this.timeUntilRegen > 10.0f && !source.isIn(DamageTypeTags.BYPASSES_COOLDOWN)
-                    ? amount - this.lastDamageTaken : amount;
+            float realAmount = this.xtimeUntilRegen > 10.0f && !source.isIn(DamageTypeTags.BYPASSES_COOLDOWN)
+                    ? amount - this.xlastDamageTaken : amount;
```

Comment:
This is really weird you have to do this, I'll check the bug you're mentioning but the solution does not look good